datageneration/gpt_data_generator.py

```diff
@@ -325,26 +333,26 @@ def generate_written_word_distance(self, metric: str, max_digits: int) -> tuple:
         :param max_digits: maximum number of digits allowed
         :return: numeric - new numeric value, written - corresponding number in written words
         """
-        digits = randint(1, max_digits-2)
-        low = np.power(10, digits - 1)
-        high = np.power(10, digits) - 1
-
-        numeric = randint(low, high) * 100
-        written = num2words(numeric) + " " + metric
-        if np.random.choice([True, False]):
-            written = written.replace(",", "")
-        numeric = str(numeric) + " " + metric
-
+        # print("===DISTANCE===")
+        # print(distance)
+        # todo: why do we generate again???? i remove it
+        # digits = randint(1, max_digits - 2)
+        # low = np.power(10, digits - 1)
+        # high = np.power(10, digits) - 1
+        #
+        # numeric = randint(low, high) * 100
+        numeric = distance
+        written = Distance(magnitude=num2words(distance.magnitude), metric=distance.metric)
+
+        # todo: I commented out, I don't know why it is important
```

Comment:
The library "num2words" will always add a comma in the written number (e.g. "two-thousand, three-hundred").. GPT had the tendency to put that comma in the sentence in all cases. This line at random removes the commas sometimes, so we have more variety in the sentences.. Better keep this line.